client/src/pages/CreditCard.js

```diff
@@ -25,18 +31,10 @@ class CreditCard extends React.Component {
             this.setState({ brand: await data['brand'] });
             this.setState({ expMonth: await data['exp_month'] });
             this.setState({ expYear: await data['exp_year'] });
-            this.setState({ ccExists: await this.state.last4 ? true : false })
-            this.setState({ ccHeader: await this.state.clientSecret ? <h4>{this.makeHeader()}</h4> : <h6>No credit card added yet</h6> })
+            this.setState({ ccExists: await this.state.last4 ? true : false });
```

Comment:
i'm not entirely sure but i believe you don't have to await all these data because once you get the response (which from reading the code, i think you are already doing a successful wait on line 27 `const response = await`). getting rid of all these `await`s should still work. 

```
const credit_card = await data; 

this.setState({
  ...
  expYear: data['exp_year'],
  ccExists: this.state.last4 ? true : false
  ...
})
```

you can put them all in one setState :) that will probably not force react to render every time!